### PR TITLE
Replace tasks table [SCI-9801]

### PR DESCRIPTION
**Pull request by aignatov-bio**
Jira ticket: [SCI-9801](https://scinote.atlassian.net/browse/SCI-9801)

### What was done
Replace tasks table 

[SCI-9801]: https://scinote.atlassian.net/browse/SCI-9801?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/services/lists/my_modules_service.rb [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+      task_comments: { user: :teams },
+      user_assignments: :user,
+      designated_users: {},
+      experiment: :project
+    }
+
+    private
+
+    def fetch_records
+      records = @raw_data.includes(PRELOAD)
```
Comment (artoscinote) [inline app/services/lists/my_modules_service.rb:18]: all these method calls would probably be nicer if records was an instance variable @records, so we don't need to pass it everywhere

File: app/javascript/packs/vue/my_modules_list.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+import PerfectScrollbar from 'vue3-perfect-scrollbar';
+import MyModulesList from '../../vue/my_modules/list.vue';
+import { mountWithTurbolinks } from './helpers/turbolinks.js';
+
+const app = createApp();
+app.component('MyModulesList', MyModulesList);
+app.config.globalProperties.i18n = window.I18n;
+app.use(PerfectScrollbar);
+window.myModulesList = mountWithTurbolinks(app, '#MyModulesList', () => {
+  window.myModulesList = null;
```
Comment (artoscinote) [inline app/javascript/packs/vue/my_modules_list.js:11]: may be more proper to use `delete window.myModulesList` here

File: app/javascript/vue/my_modules/list.vue [hunk 3]
```diff
@@ -0,0 +1,337 @@
...
+      }
+
+      return {
+        left,
+        right: []
+      };
+    }
+  },
+  methods: {
+    usersFilterRenderer(option) {
```
Comment (artoscinote) [inline app/javascript/vue/my_modules/list.vue:252]: would be nicer to group the renderer functions together, maybe at the end of the methods object

File: app/javascript/vue/my_modules/modals/edit.vue [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+  props: {
+    my_module: Object
+  },
+  data() {
+    return {
+      name: this.my_module.name
+    };
+  },
+  computed: {
+    validName() {
```
Comment (artoscinote) [inline app/javascript/vue/my_modules/modals/edit.vue:49]: i think we should start sticking to some JS naming conventions, in this case (a boolean) it should be called `nameIsValid`

File: app/serializers/lists/my_module_serializer.rb [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+
+    def status
+      {
+        name: object.my_module_status.name,
+        color: object.my_module_status.color,
+        light_color: object.my_module_status.light_color?
+      }
+    end
+
+    def assigned_users
```
Comment (artoscinote) [inline app/serializers/lists/my_module_serializer.rb:123]: **_designated_** users!

File: app/services/lists/my_modules_service.rb [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+
+    def archived_on_from_filter(my_modules, value)
+      my_modules.where('my_modules.archived_on >= ?', value)
+    end
+
+    def archived_on_to_filter(my_modules, value)
+      my_modules.where('my_modules.archived_on <= ?', value)
+    end
+
+    def assigned_users_filter(my_modules, users)
```
Comment (artoscinote) [inline app/services/lists/my_modules_service.rb:98]: **_designated_** users!

File: app/javascript/vue/my_modules/renderers/assigned_users.vue [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+  </GeneralDropdown>
+</template>
+
+<script>
+
+import axios from '../../../packs/custom_axios.js';
+import GeneralDropdown from '../../shared/general_dropdown.vue';
+
+export default {
+  name: 'AssignedUsersRenderer',
```
Comment (artoscinote) [inline app/javascript/vue/my_modules/renderers/assigned_users.vue:49]: **_designated_** users!

**Review by artoscinote — CHANGES_REQUESTED** [review]
Please be careful not to mix designated_users (user_my_modules) and assigned_users (user_assignments)

**Review by rekonder — APPROVED** [review]

**Pull request merged**